### ISSUE TITLE
RS1024 does not trigger on calls to List<T>.Contains

### ISSUE THREAD

**Issue by bjornhellander** [dotnet/roslyn-analyzers#7610]
### Analyzer

**Diagnostic ID**: RS1024

### Analyzer source

**NuGet Package**: Microsoft.CodeAnalysis.Analyzers

**Version**: 3.12.0-beta1.25081.1

### Describe the bug

RS1024 is not triggered on calls to `List<T>.Contains`, but I think it should.

### Steps To Reproduce

```
private void M(List<ITypeSymbol> types1, IReadOnlyList<ITypeSymbol> types2, ITypeSymbol type)
{
    _ = types1.Contains(type); // No RS1024!
    _ = types2.Contains(type); // RS1024
}
```

### Expected behavior

RS1024 should trigger on both calls to Contains.

### Actual behavior

It only triggers on the second call.
